Library/Homebrew/brew.sh

```diff
@@ -179,6 +179,15 @@ then
   set -- "$@" -v
 fi
 
+# Check if user passes --help command which should prevent auto-update.
+for arg in "$@"
+do
+  if [[ $arg = "--help" ]]
```

Comment:
Seems like even more than that. https://github.com/Homebrew/brew/blob/master/Library/Homebrew/brew.rb#L34
```ruby
 help_flag_list = %w[-h --help --usage -?]
```